### ISSUE TITLE
date/dateTime Validation

### ISSUE THREAD

**Issue by GaryGSC** [byu-oit/openapi-enforcer#3]
The `date` and `dateTime` validation currently seems wrong. Both validation functions expect a Javascript Date Object and consider strings to be errors (even when formatted correctly). For an OpenAPI document that specifies a (formatted) string to be the correct type, this feels off.

This results in errors like:
```
Response invalid:
  Error in body/date_time_updated/value: Expected a valid date object. Received: '2017-03-30T08:46:16.000Z'
```

This isn't terrible, but it complicates mocking, since my examples (strings in a JSON file) are considered invalid.

**Comment by GaryGSC** [comment]
I now have another example of running into this issue:

I have one service using `openapi-enforcer-middleware` (**A**) that acts as a thin facade in front of another service (**B**). I intend to have **A** return to the caller exactly what the **B** gave it, but part of what **B** returns has dates and dateTimes as strings. To appease the validator, **A** needs to go through each of the dates and dateTimes in **B**'s response to convert them to Javascript Date objects.

**Comment by GaryGSC** [comment]
Resolved (at least for my use cases) in a8fd30111f2030b73c9a4b5fee9a962708a4ac04. Thanks!
